numpy/core/src/multiarray/descriptor.c

```diff
@@ -1493,19 +1493,47 @@ _convert_from_any(PyObject *obj, int align)
         return ret;
```

Comment:
Line 1491 needs a recursion check too

Reply:
Oh sorry for missing this, will add

Reply:
On the same note, what is a way to trigger recursion limit with str, I want to add UT for it. @eric-wieser

Reply:
Actually, I don't think this case can ever happen - so you can remove the recursion checks for string and for bytes